OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/fieldassignability/AbstractFieldAssignabilityAnalysis.scala

```diff
@@ -128,29 +128,34 @@ trait AbstractFieldAssignabilityAnalysis extends FPCFAnalysis {
 
         implicit val state: AnalysisState = createState(field)
 
-        if (field.isFinal)
-            return Result(field, NonAssignable);
-        else
-            state.fieldAssignability = EffectivelyNonAssignable
+        val fieldName = field.name
+        println(fieldName)
+
+        state.fieldAssignability =
+            if (field.isFinal)
+                NonAssignable
+            else
+                EffectivelyNonAssignable
 
-        if (field.isPublic)
+        if (field.isPublic && !field.isFinal)
```

Comment:
Wouldn't it be easier and more readable to just wrap the entire section into a separate `if (!field.isFinal)`?